zilencer/auth.py

```diff
@@ -37,6 +39,29 @@
 
 ParamT = ParamSpec("ParamT")
 
+REMOTE_SERVER_TAKEOVER_TOKEN_SALT = "remote_server_takeover"
+REMOTE_SERVER_TAKEOVER_TOKEN_VALIDITY_SECONDS = 300
+
+
+def generate_registration_takeover_verification_secret(hostname: str) -> str:
+    signer = TimestampSigner(salt=REMOTE_SERVER_TAKEOVER_TOKEN_SALT)
+    secret = base64.b16encode(signer.sign(hostname).encode()).decode()
+    return secret
+
+
+def verify_registration_takeover_verification_secret(secret: str, hostname: str) -> bool:
+    signer = TimestampSigner(salt=REMOTE_SERVER_TAKEOVER_TOKEN_SALT)
+    try:
+        signed_data = base64.b16decode(secret).decode()
+        hostname_from_secret = signer.unsign(
+            signed_data, max_age=REMOTE_SERVER_TAKEOVER_TOKEN_VALIDITY_SECONDS
+        )
+    except BadSignature:
+        return False
+    if hostname_from_secret != hostname:
+        return False
+    return True
```

Comment:
We might want to rework this to `raise JsonableError` here, rather than returning boolean, just to propagate back to the caller an error message for the specific case that the hostname is wrong? I feel like that's a failure mode that will probably happen.

Reply:
Done, distinct errors now:
```
    except SignatureExpired:
        raise JsonableError(_("The verification secret has expired"))
    except BadSignature:
        raise JsonableError(_("The verification secret is invalid"))
    if hostname_from_secret != hostname:
        raise JsonableError(_("The verification secret is for a different hostname"))
```

And to the user of the command it looks like:
```
Error received from the push notification service: "The verification secret has expired"
```